extensions/toggl-track/src/components/EditTimeEntryForm.tsx

```diff
@@ -0,0 +1,103 @@
+import {
+  useNavigation,
+  Form,
+  ActionPanel,
+  SubmitFormAction,
+  Icon,
+  showToast,
+  ToastStyle,
+  clearSearchBar,
+} from "@raycast/api";
+import toggl from "../toggl";
+import { storage } from "../storage";
+import { Project, TimeEntry } from "../toggl/types";
+import { useAppContext } from "../context";
+import { useMemo, useState } from "react";
+
+function EditTimeEntryForm({ project, entry }: { project?: Project; entry: TimeEntry }) {
+  const navigation = useNavigation();
+  const { projects, tags, isLoading, projectGroups } = useAppContext();
+  const [selectedProject, setSelectedProject] = useState<Project | undefined>(project);
+  const [selectedTags, setSelectedTags] = useState<string[]>([]);
+
+  async function handleSubmit(values: { description: string; tags: string[]; start: string; end?: string}) {
+    try {
+      await toggl.editTimeEntry({
+        id: entry.id,
+        projectId: selectedProject?.id,
+        description: values.description,
+        tags: values.tags,
+        start: values.start,
+        duration: (values.end ? Math.trunc((new Date(values.end).getTime() - new Date(values.start).getTime()) / 1000) : -1)
+      });
+      await showToast(ToastStyle.Animated, "Updating time entry...");
+      "stop" in entry ? await storage.timeEntries.refresh() : await storage.runningTimeEntry.refresh();;
+      await showToast(ToastStyle.Success, "Updated time entry");
+      navigation.pop();
+      await clearSearchBar();
+    } catch (e) {
+      await showToast(ToastStyle.Failure, "Failed to update time entry");
+    }
+  }
+
+  const projectTags = useMemo(() => {
+    return tags.filter((tag) => tag.wid === selectedProject?.wid);
+  }, [tags, selectedProject]);
+
+  const onProjectChange = (projectId: string) => {
+    const project = projects.find((project) => project.id === parseInt(projectId));
+    if (project) {
+      setSelectedProject(project);
+    }
+  };
+
+  const onTagsChange = (tags: string[]) => {
+    setSelectedTags(tags);
+  };
+
+  return(
+    <Form
```

Comment:
This form should be extracted to its own component, so the interaction is the same for creating time entries and editing time entries